### PR TITLE
Upgrade to asciidoctorj 2.x and fix extension registry. Fixes #1

### LINKED ISSUE #1: Does not work with Asciidoctorj 2.x
It looks like `org.asciidoctor.extension.spi.ExtensionRegistry` has been relocated to `org.asciidoctor.jruby.extension.spi.ExtensionRegistry` in Asciidoctorj 2.x. This makes version `0.2` of this extension miss the required hook to perform its work.

### PR DESCRIPTION
**Pull request by aalmiray**
Upgrades dependencies to Asciidoctorj 2.x and fixes the wrong path to the extension registry.
This patch makes the extension work with Asciidoctorj 2.x and breaks compatibility with Asciidoctorj 1.6.0.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -5,7 +5,7 @@ plugins {
 }
 
 group = "com.bmuschko"
-version = "0.2"
+version = "0.3-SNAPSHOT"
```
Comment (bmuschko) [inline build.gradle.kts]: Let's revert this change. Version 0.2 hasn't been released yet so the version has to be 0.2.

**Review by bmuschko — CHANGES_REQUESTED** [review]
The documentation of this project will have to indicate the compatible minimal version of AsciidoctorJ for a potential 0.2 version. See "Important" box in README.

File: build.gradle.kts (hunk 1 above)
Comment (aalmiray) [reply build.gradle.kts]: That’s odd, version 0.2 is visible from jcenter for quite a while

Comment (bmuschko) [reply build.gradle.kts]: Looks like I didn't update the documentation. My bet. So yeah, the next version is going to be 0.3. Can you either change this to 0.3? I am not going to publish a SNAPSHOT.

**Comment by bmuschko** [comment]
Before merging, could you please also update the "Important" note spelling out the minimal compatible version of AsciidoctorJ?

**Review by aalmiray — COMMENTED** [review]
Updated readme and version as requested

**Review by bmuschko — APPROVED** [review]
